megatron/core/resharding/refit.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+"""
+High-level refit/reshard orchestration:
+- swap_model_weights: public API; accepts a backend name or CopyService and delegates.
+- reshard_model_weights: transport-agnostic core; builds/caches plan and executes.
+"""
+
+from typing import Any, Optional, Union
+
+from megatron.core.models.common.language_module.language_module import LanguageModule
+from megatron.core import parallel_state
+from . import build_centralized_reshard_plan, execute_reshard_plan
+from .copy_services.base import CopyService
+from .copy_services.nccl_copy_service import NCCLCopyService
+
+
+def _unwrap_module(module: LanguageModule) -> Any:
```

Comment:
Would megatron.core.utils.unwrap_model work for you? 

https://github.com/NVIDIA/Megatron-LM/blob/ffb8c357e594f2eb14e71f6ce8831c0ffaf97f13/megatron/core/utils.py#L2055C5-L2055C17